### PR TITLE
Single Node deployment with bootstrap-in-place

### PR DESCRIPTION
**Pull request by eranco74**
### Adding bootstrap-in-place functionality
Add a new entrypoint `--bootstrap-in-place` which enrich the master
Ignition with control plane static pods manifests and all required
resources, including the etcd database.

`cluster-bootstrap` normally waits for a list of required pods to be
ready. These pods are expected to start running on the control plane
nodes when the bootstrap and control plane run in parallel. That is
not possible when bootstrapping in place, so when `cluster-bootstrap`
runs with the `--bootstrap-in-place` option it should only apply the
manifests and then tear down the control plane.

If `cluster-bootstrap` fails to apply some of the manifests, it should
return an error.

### CODE REVIEW

**Comment by romfreiman** [comment]
Ship it :)

File: pkg/start/bootstrap.go [hunk 1]
```diff
@@ -66,26 +66,23 @@ func (b *bootstrapControlPlane) waitForApi() error {
...
+			previousError = err.Error()
 		}
-	}, time.Second, apiContext.Done())
-	if apiContext.Err() == context.Canceled {
-		return nil
-	} else {
+
+		return false, nil
+	}, apiContext.Done())
+	if err != nil {
```
Comment (sttts) [inline pkg/start/bootstrap.go:81]: these changes should be squashed into first commit

**Comment by sttts** [comment]
/hold 

until master reopen for 4.8.

/approve
LGTM up to the squashing.

**Pull request merged**